### PR TITLE
Extend NodeAssociations object to include peer-source associations

### PR DESCRIPTION
**Pull request by RVanhuysseXenit**
Fixes https://xenitsupport.jira.com/browse/ALFREDAPI-420

- [x] Is [CHANGELOG.md](https://github.com/xenit-eu/alfred-api/blob/master/CHANGELOG.md) extended?
- [x] Does this PR avoid breaking the API? 
    Breaking changes include adding, changing or removing endpoints and/or JSON objects used in requests and responses.
- [x] Does the PR comply to REST HTTP result codes policy outlined in the [developer guide](https://github.com/xenit-eu/alfred-api/blob/master/developer-documentation)?
- [x] Does the PR follow our [coding styleguide and other active procedures](https://xenitsupport.jira.com/wiki/spaces/XEN/pages/624558081/XeniT+Enhancement+Proposals+XEP)?
- [x] Is usage of `this.` prefix avoided?

See [README.md](./README.md) for full explanation.

### CODE REVIEW

**Comment by tgeens** [comment]
Why are we not using proper PR titles ?
Why do we have this PR template that is not used anyway ?
Why do we manage issues in private for an open source component ?

**Comment by RVanhuysseXenit** [comment]
@tgeens 
* Because I thought it had taken over the commit title, which is better. Though it also starts with the ticket so I missed it did not do this
* I did fill in the checkmarks in the template, but did it incorrect which broke the template. its now as it should be
* That relates to the discussion on open/closed issuetrackers and CI, in which I have no say. The current MO is a closed tracker. Issues should still be referenced in relevant prs.

File: apix-rest-v1/src/main/java/eu/xenit/apix/rest/v1/nodes/NodesWebscript1.java [hunk 1]
```diff
@@ -406,7 +406,8 @@ public void getAllInfoOfNode(@UriVariable String space, @UriVariable String stor
                     +
                     "Set 'retrieveChildAssocs' to false to omit the child associations from the result.\n" +
                     "Set 'retrieveParentAssocs' to false to omit the parent associations from the result.\n" +
-                    "Set 'retrieveTargetAssocs' to false to omit the peer associations from the result.\n")
+                    "Set 'retrieveTargetAssocs' to false to omit the peer target associations from the result.\n" +
+                    "Set 'retrieveSourceAssocs' to false to omit the peer source associations from the result.\n")
```
[the remark this answers is not in the document]
Comment (WimCrols) [reply apix-rest-v1/src/main/java/eu/xenit/apix/rest/v1/nodes/NodesWebscript1.java:410]: For /sites/mySites the default is false, so there there is no problem.

**Comment by WimCrols** [comment]
@tgeens 
I don't understand your first two points, but it looks like they are fixed? 

The OSS point is because we need tickets that we can manage on our board and book time, but in all 
honesty, restarting the OSS discussion should happen in a meeting with all stakeholders and not in a PR :)

File: apix-impl/src/main/java/eu/xenit/apix/alfresco/metadata/NodeService.java [hunk 2]
```diff
@@ -373,54 +374,58 @@ private boolean containsAspect(List<AspectDefinition> aspects, AspectDefinition
...
-            ChildParentAssociation apixChildAssoc = new ChildParentAssociation(ref, c.apix(alfChildRef),
-                    c.apix(alfType), isPrimary);
-            apixChildAssocs.add(apixChildAssoc);
-        }
-
-        return apixChildAssocs;
+        return nodeService.getChildAssocs(c.alfresco(ref))
+                .stream().map(alfChildAssoc ->
+                        new ChildParentAssociation(
+                                ref,
```
Comment (RVanhuysseXenit) [inline apix-impl/src/main/java/eu/xenit/apix/alfresco/metadata/NodeService.java:380]: @WimCrols I want to note that there might be a semantic error here (also in the original) in that both getParent and getChild return association objects in which the target noderef is listed as the source of the returned association. This means that in one case (imo the getChild) the direction of the returned association is inverted?
Maybe we should open another ticket for it?

File: apix-rest-v1/src/main/java/eu/xenit/apix/rest/v1/ApixV1Webscript.java [hunk 3]
```diff
@@ -82,7 +84,7 @@ protected NodeInfo nodeRefToNodeInfo(NodeRef nodeRef, IFileFolderService fileFol
             INodeService nodeService, IPermissionService permissionService) {
         return nodeRefToNodeInfo(nodeRef, fileFolderService, nodeService, permissionService,
                 true, true, true, true,
-                true, true, true);
+                true, true, true, true);
```
[the remark this answers is not in the document]
Comment (driesdeproost) [reply apix-rest-v1/src/main/java/eu/xenit/apix/rest/v1/ApixV1Webscript.java:87]: image

**Comment by tgeens** [comment]
The first two are fixed now, the 3rd was a personal annoyance :-)

File: apix-impl/src/main/java/eu/xenit/apix/alfresco/metadata/NodeService.java (hunk 2 above)
Comment (WimCrols) [reply apix-impl/src/main/java/eu/xenit/apix/alfresco/metadata/NodeService.java:380]: Good idea to open another ticket.

**Pull request merged**